lib/ClangImporter/ImportType.cpp

```diff
@@ -1806,8 +1806,20 @@ ParameterList *ClangImporter::Implementation::importFunctionParameterList(
     // Import the parameter type into Swift.
     Type swiftParamTy;
     bool isParamTypeImplicitlyUnwrapped = false;
-    if (auto *templateParamType =
-            dyn_cast<clang::TemplateTypeParmType>(paramTy)) {
+
+    if (isa<clang::ReferenceType>(paramTy) &&
```

Comment:
Nit: use `if (auto referenceType = dyn_cast<...>`? for both referenceType and for templateParamType? I think it will make the body more readable at the price of increased indent.

Reply:
The only issue is then the `else if` doesn't work. I could introduce a variable `isReferenceWrappedTempalte` but that basically has the same issue as the current if statement. Happy to do that if you'd rather, though.

Reply:
I've replaced the calls to `isa<clang::ReferenceType>` and `cast<clang::ReferenceType>` with a `dyn_cast<clang::ReferenceType>` at the top of the if. Let me know what you think.

Reply:
Ah I see, yeah then it cannot be super clean anyway. The code we how now LGTM. Thanks!